packages/react-aria-components/stories/Table.stories.tsx

```diff
@@ -392,3 +393,134 @@ const MyCheckbox = ({children, ...props}: CheckboxProps) => {
     </Checkbox>
   );
 };
+
+const MyTableLoader = () => {
+  return (
+    <UNSTABLE_TableLoader>
+      <span>
+        Load more spinner
+      </span>
+    </UNSTABLE_TableLoader>
+  );
+};
+
+function MyTableBody(props) {
+  let {rows, children, isLoading, ...otherProps} = props;
+  return (
+    <TableBody {...otherProps}>
+      <Collection items={rows}>
+        {children}
+      </Collection>
+      {isLoading && <MyTableLoader />}
+    </TableBody>
+  );
+}
+
+const TableLoadingBodyWrapper = (args: {isLoading: boolean}) => {
+  return (
+    <Table aria-label="Files">
+      <TableHeader columns={columns}>
+        {(column) => (
+          <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+        )}
+      </TableHeader>
+      <MyTableBody rows={rows} isLoading={args.isLoading}>
+        {(item) => (
+          <Row columns={columns}>
+            {(column) => {
+              return <Cell>{item[column.id]}</Cell>;
+            }}
+          </Row>
+        )}
+      </MyTableBody>
+    </Table>
+  );
+};
+
+export const TableLoadingBodyWrapperStory = {
+  render: TableLoadingBodyWrapper,
+  args: {
+    isLoading: false
+  },
+  name: 'Table loading, table body wrapper with collection'
+};
+
+function MyRow(props) {
+  return (
+    <>
+      {/* Note that all the props are propagated from MyRow to Row, ensuring the id propagates */}
+      <Row {...props} />
+      {props.isLoading && <MyTableLoader />}
+    </>
+  );
+}
+
+const TableLoadingRowRenderWrapper = (args: {isLoading: boolean}) => {
+  return (
+    <Table aria-label="Files">
+      <TableHeader columns={columns}>
+        {(column) => (
+          <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+        )}
+      </TableHeader>
+      <TableBody items={rows} dependencies={[args.isLoading]}>
+        {(item) => (
+          <MyRow columns={columns} isLoading={item.id === 4 && args.isLoading}>
+            {(column) => {
+              return <Cell>{item[column.id]}</Cell>;
+            }}
+          </MyRow>
+        )}
+      </TableBody>
+    </Table>
+  );
+};
+
+export const TableLoadingRowRenderWrapperStory = {
+  render: TableLoadingRowRenderWrapper,
+  args: {
+    isLoading: false
+  },
+  name: 'Table loading, row renderer wrapper and dep array'
+};
+
+
+function renderEmptyLoader({isLoading}) {
+  return isLoading ? 'Loading spinner' : 'No results found.';
+}
+
+const RenderEmptyState = (args: {isLoading: boolean}) => {
+  let {isLoading} = args;
+  return (
+    <Table aria-label="Files" selectionMode="multiple">
+      <TableHeader columns={columns}>
+        <Column><MyCheckbox slot="selection" /></Column>
+        <Collection items={columns}>
+          {(column) => (
+
+            <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+          )}
+        </Collection>
+      </TableHeader>
+      <TableBody renderEmptyState={() => renderEmptyLoader({isLoading})}>
```

Comment:
The idea now is that the user will handle the empty state and the "isLoading" state (aka no items but is loading) via renderEmptyState, no longer using UNSTABLE_TableLoader for this. Rendering the "loadMore" state (aka items exist but is loading) is the responsibility of the user where they should only include the UNSTABLE_TableLoader when they are actually performing load more and not when the table is still empty. 

This simplifies the collections changes we need to do since the way to compute collection size remains the same